Feature/StoryLog 이전 대화 내용 기록 기능 추가

Kkumteulyi opened the pull request:
## 개요
<!-- 어떤 점이 변경되었는지, 간단하게 작성해주세요. -->
스토리 진행 도중에 나왔던 대사들을 확인할 수 있는 기능을 추가하였습니다.

## 변경 사항
<!-- 어떤 점에 변경되었는지 상세하게 작성해주세요.-->
### 이전 대화 내용 기록 
- 최대 30개의 대화를 기록하는 UI와 패널을 열 수 있는 버튼을 추가하였습니다
- 가장 최근에 추가된 대화부터 아래에서 위로 쌓여가는 방식으로 구현되어 있습니다
- 현재 이벤트에 등장하는 대화를 기록하고 해당 이벤트가 종료되면 기록도 함께 삭제됩니다

### 함수
**OnPointerDown**
- 클릭 이벤트 발생 시 대화 기록 패널의 활성화 여부에 따라 대화 기록을 켜고 끕니다
- 대화 기록 창이 활성화되어 있다면, 버튼의 글자를 노란색으로 표시합니다

**OnPointerEnter**
- 마우스의 위치가 해당 버튼 위에 존재할 경우 버튼에 눌림 효과를 주기 위해 글자의 색을 어둡게 변경합니다

**OnPointerExit**
- 마우스의 위치가 버튼을 벗어나면 원래 버튼의 색으로 변경합니다.

**AddLog**
- DialogueManager에서 이름과 대화 내용을 가져와 각각의 컴포넌트에 할당하는 함수입니다.
- 프리팹 오브젝트들에서 GetChild를 사용하여 이름, 대화 내용 오브젝트에 접근 후, GetComponent를 통해 TMP_Text에 할당합니다

**ResetLogs**
- 30개의 로그들을 전부 비활성화 시킵니다
- 각각의 로그에 해당하는 이름과 대화를 저장한 변수들을 초기화시킵니다

## 참고 자료
<!-- 기능 개발에 참고한 자료가 있다면 작성해주세요. (필수는 아닙니다.) -->
자식 오브젝트의 컴포넌트 가져오기 - https://learnandcreate.tistory.com/2210
동일한 태그를 가진 오브젝트를 배열로 만들기 - https://curioso365.tistory.com/106

Kkumteulyi committed:
대화 기록 틀 제작

Kkumteulyi committed:
24.07.09 - 대화 기록창 구현 완료

대화 기록 UI
- 대화의 이름, 대사를 기록하는 기능 추가
- 최대 10개 까지의 기록을 확인 가능

Kkumteulyi committed:
로그 출력 파트 수정

로그 출력 순서 수정
- 기존에 무작위로 로그가 출력되던 오류 수정

로그 UI 변경
- 버튼 클릭 시 불투명한 막 활성화
- 로그 박스(로그가 적힌 오브젝트)들이 화면 정중앙에 오도록 수정
- 글자 크기 증가 및 왼쪽 정렬로 변경
- 버튼 클릭 시 버튼 이미지 변경 효과 추가

miniron-v reviewed (changes requested):
2가지 정도 추가 수정 사항이 있네요.

## 1. 로그 창 우선순위 문제
![image](https://github.com/user-attachments/assets/3297ad43-d7fc-4269-ac12-b9e54c6117d3)

로그 창이 **선택지를 가리는 게** 일반적일 듯 한데, 지금은 반대로 되어 있네요.

## 2. 이벤트 종료 후에도 로그가 남아 있는 문제
한 이벤트가 종료되고 나면, 해당 로그들은 사라지는 게 맞다고 생각합니다.

Kkumteulyi committed:
Merge branch 'develop' into feature/StoryLog

Kkumteulyi committed:
수정 사항 반영

Develope에서 머지 이후 기존 수정 사항들 이동

Kkumteulyi committed:
Merge branch 'develop' into feature/StoryLog

Kkumteulyi committed:
대화 기록 초기화 기능, 기록 가능한 대화수 확장, 기타 사항

대화 기록 초기화
- 이벤트가 종료되면 해당 이벤트에서 등장한 대화는 삭제

기록 가능한 대화 수 확장
- 기록이 10개를 넘어가면 삭제되던 방식에서 이벤트가 바뀔 때 마다 해당 기록들이 삭제되도록 방식 변경
- 최대 30개의 기록이 나오도록 수정

기타 사항
- 대화 기록 버튼에 마우스 커서에 따른 글자 색 변경 효과 추가

Kkumteulyi commented:
위의 2가지 사항 수정하였습니다.

miniron-v committed:
Merge branch 'develop' into feature/StoryLog
